core/src/bitmap/turbulence.rs

```diff
@@ -229,14 +229,15 @@ impl Turbulence {
             let noise = self.noise2(color_channel, vec, stitch_info);
             sum += if fractal_sum { noise } else { noise.abs() } / ratio;
             ratio *= 2.0;
-            stitch_info.as_mut().map(|stitch_info| {
+            stitch_info = stitch_info.map(|si| {
```

Comment:
Can't we do the following here?

```rust
stitch_info.as_mut().inspect(|stitch_info| {
```

Reply:
That's what I tried first, but couldn't find a way to allow the closure to mutate its parameter. And exactly because it mutates, wouldn't it be a bit confusing that way?

Reply:
Oh, you're right, I missed that. I believe we can do the following then, which is IMO more readable.
```rust
if let Some(ref mut stitch_info) = stitch_info {
    // ...
}
```

Because otherwise we're unnecessarily constructing a new object.

Reply:
Oh, yep, that works. Thanks!